### DISCUSSION TITLE
Master commit hash on a pull request action

### DISCUSSION THREAD

**Discussion by beldar** [community/community#26506] [Actions]
<p>Hi there,</p>
<p>Does anyone know if there’s a way to get the commit hash for the current head of the master branch on an action that runs on a pull_request?</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/beldar">@beldar</a> ,</p>
<p>You can use the property " <strong>github.event.pull_request.head.sha</strong>" from the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> to fetch the commit hash for the PR’s <strong>head</strong>.<br>
Similarly, you also can use " <strong>github.event.pull_request.base.sha</strong>" to fetch the commit hash for the PR’s <strong>base</strong> (master in your case).</p>
<p>You can print the <strong>github context</strong> information to the logs to fetch more detailed information about the PR.</p>
<pre><code>- name: Dump GitHub context
  env:
    GITHUB_CONTEXT: ${{ toJson(github) }}
  run: echo "$GITHUB_CONTEXT"
</code></pre>

**Comment by beldar** [comment]
<p>That’s great! Exactly what I needed.</p>
<p>Thanks!!</p>
